storage/src/main/java/tech/pegasys/artemis/storage/server/rocksdb/core/RocksDbInstanceFactory.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.artemis.storage.server.rocksdb.core;
+
+import com.google.common.collect.ImmutableMap;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import org.apache.tuweni.bytes.Bytes;
+import org.rocksdb.BlockBasedTableConfig;
+import org.rocksdb.ColumnFamilyDescriptor;
+import org.rocksdb.ColumnFamilyHandle;
+import org.rocksdb.ColumnFamilyOptions;
+import org.rocksdb.DBOptions;
+import org.rocksdb.Env;
+import org.rocksdb.LRUCache;
+import org.rocksdb.RocksDBException;
+import org.rocksdb.TransactionDB;
+import org.rocksdb.TransactionDBOptions;
+import tech.pegasys.artemis.storage.server.DatabaseStorageException;
+import tech.pegasys.artemis.storage.server.rocksdb.RocksDbConfiguration;
+import tech.pegasys.artemis.storage.server.rocksdb.schema.RocksDbColumn;
+import tech.pegasys.artemis.storage.server.rocksdb.schema.Schema;
+
+public class RocksDbInstanceFactory {
+  static {
+    RocksDbUtil.loadNativeLibrary();
+  }
+
+  private final ColumnFamilyHandle defaultHandle;
+  private final ImmutableMap<RocksDbColumn<?, ?>, ColumnFamilyHandle> columnHandles;
+  private final TransactionDB db;
+  private final List<AutoCloseable> resources = new ArrayList<>();
+
+  public static RocksDbInstance create(
+      final RocksDbConfiguration configuration, final Class<? extends Schema> schema)
+      throws DatabaseStorageException {
+    final RocksDbInstanceFactory factory = new RocksDbInstanceFactory(configuration, schema);
+    return new RocksDbInstance(
+        factory.db, factory.defaultHandle, factory.columnHandles, factory.resources);
+  }
```

Comment:
It feels weird to have a static `create` but then do all the work in the constructor and ultimately throw away the constructed object.  It looks like we could just make all the methods static and move the constructor into `create` fairly easily.

Reply:
I originally had this as an instance inside of `RocksDbInstance` and I just didn't fully convert it to use static methods - will clean it up 👍